ludwig/api.py

```diff
@@ -1938,39 +1939,53 @@ def upload_to_hf_hub(
 
         # Inputs
 
-        :param repo_id (`str`):
+        :param repo_id: (`str`)
             A namespace (user or an organization) and a repo name separated
             by a `/`.
-        :param model_path (`str`):
-            The path of the saved model. This is the top level directory where
-            the models weights as well as other associated training artifacts
-            are saved.
-        :param private (`bool`, *optional*, defaults to `False`):
+        :param model_path: (`str`)
+            The path of the saved model. This is either (a) the directory where
+            the 'models_weights' folder and the 'model_hyperparameters.json' file
+            are stored, or (b) the parent of that folder.
+        :param private: (`bool`, *optional*, defaults to `False`)
             Whether the model repo should be private.
-        :param repo_type (`str`, *optional*):
+        :param repo_type: (`str`, *optional*)
             Set to `"dataset"` or `"space"` if uploading to a dataset or
             space, `None` or `"model"` if uploading to a model. Default is
             `None`.
-        :param commit_message (`str`, *optional*):
+        :param commit_message: (`str`, *optional*)
             The summary / title / first line of the generated commit. Defaults to:
             `f"Upload {path_in_repo} with huggingface_hub"`
-        :param commit_description (`str` *optional*):
+        :param commit_description: (`str` *optional*)
             The description of the generated commit
 
         # Returns
 
         :return: (bool) True for success, False for failure.
         """
+        if os.path.exists(os.path.join(model_path, "model", "model_weights")) and os.path.exists(
```

Comment:
@sanjaydasgupta The idea/direction seems great!  I just have a couple of minor suggestions for you to consider (because if they work, they might improve the code quality slightly):
1) Can we create the constants for the two directory names, `"model"` and `"model_weights"` and replace codebase-wise?
2) Is it possible instead of passing the boolean `model_path_is_experiment_path`, determine the "experiments" directory and standardize all methods on using it -- so the "experiment" directory would be passed to them.  I think that even if a function is passed the "model" directory, and in those cases we have to make the assumption that "experiments" is "model/..", this would still be reasonable and acceptable.  What do you think?

Thank you very much!

Reply:
@alexsherstinsky, responding to your (2) above - sounds like a great idea! Let me take a closer look at the code.

Reply:
> @sanjaydasgupta The idea/direction seems great! I just have a couple of minor suggestions for you to consider (because if they work, they might improve the code quality slightly):
> 
> 1. Can we create the constants for the two directory names, `"model"` and `"model_weights"` and replace codebase-wise?
> 2. Is it possible instead of passing the boolean `model_path_is_experiment_path`, determine the "experiments" directory and standardize all methods on using it -- so the "experiment" directory would be passed to them.  I think that even if a function is passed the "model" directory, and in those cases we have to make the assumption that "experiments" is "model/..", this would still be reasonable and acceptable.  What do you think?
> 
> Thank you very much!

Hi @alexsherstinsky, both your suggestions have been implemented.

There was already a label `MODEL_WEIGHTS_FILE_NAME` defined (and used in some places) for the literal string "model_weights". So the choice of a label for "model" fell naturally on `MODEL_FILE_NAME` -- not the best, but there it is for the time being.